src/data/fintype/intervals.lean

```diff
@@ -49,6 +49,19 @@ begin
 end
 lemma Ioc_ℤ_finite (l u : ℤ) : set.finite (Ioc l u) := Icc_ℤ_finite (l + 1) u
 
+instance (l u : ℤ) : fintype (Ioo l u) :=
+fintype.of_finset (finset.Ico_ℤ (l+1) u) $ λ n,
+  by simp only [mem_Ioo, finset.Ico_ℤ.mem, int.add_one_le_iff, iff_self, implies_true_iff]
+
+instance (l u : ℤ) : fintype (Icc l u) :=
```

Comment:
I think you should be able to use this to prove `Icc_ℤ_finite` above

Reply:
Not in a computable way, I think.